packages/programs/src/program.ts

```diff
@@ -0,0 +1,34 @@
+import type { Address } from '@solana/addresses';
+
+/**
+ * Defines a Solana program.
+ */
+export type Program<TAddress extends string = string> = {
+    /**
+     * A unique name for the Program.
+     *
+     * To avoid conflict with other organizations, it is recommended
+     * to prefix the program name with a namespace that is unique to
+     * your organization. For instance, programs belonging to the
+     * Solana Program Library are prefixed with `spl` like so:
+     * `splMemo` or `splToken`.
+     */
+    name: string;
```

Comment:
What's this used for? Is it just intended for debugging/logging?

Reply:
Good question. I debated whether to keep that one but that's something we get for free from the IDL so it could be nice to keep in the definition.

The main reason I used the `name` attribute with the Umi framework is that it would allow you to search a program by name or program address. It's not uncommon for program developers to deploy their programs using different addresses on different clusters. In this case, a search by name would allow you to find all instances of them.

In the Umi framework though, [the program definition](https://github.com/metaplex-foundation/umi/blob/8bbb100378b578c108d7b0e49bc3f8f6773be02e/packages/umi/src/Program.ts#L58-L65) also includes an `isOnCluster` function that allows us to filter programs even further. So with Umi you could do `umi.programs.get('myProgramName', 'devnet')` and it would find you the correct program in the correct cluster even when program addresses differ on different clusters.

That being said, I'm happy to remove it from this definition of a `Program`. We could always add a `ProgramWithName` interface if required in the future.

Reply:
It definitely seems like it could be useful for debugging etc, much nicer than an address!

If we use the name from the IDL though, is that likely to have the prefix the comment mentions?

Reply:
It depends on how the IDL is generated but with metaplex for instance, we used to always name our programs `mpl_my_program` so that worked well.

In any case, you can easily update the name of your programs in your `kinobi.cjs` file to whatever you want.

We might want to consider offering an even stronger naming convention while we're at it though. Maybe even a domain based on like `com.solana.system`?